tokio/src/util/blocking_check.rs

```diff
@@ -0,0 +1,25 @@
+#[cfg(unix)]
+use std::os::fd::AsFd;
+
+#[cfg(unix)]
+#[allow(unused_variables)]
+pub(crate) fn check_socket_for_blocking<S: AsFd>(s: &S) -> crate::io::Result<()> {
+    #[cfg(debug_assertions)]
+    {
+        let sock = socket2::SockRef::from(s);
+
+        if !sock.nonblocking()? {
+            eprintln!("Warning: registering a blocking socket, this may be a bug!");
```

Comment:
The warning should say that it comes from Tokio. Perhaps include `Location::caller` via `#[track_caller]` to print the caller location.